### ISSUE TITLE
Build: Cannot compile modules unless the '--module' flag is provided.

### ISSUE THREAD

**Issue by flyfishMT** [Microsoft/TypeScript#4300]
I get this error - Build: Cannot compile modules unless the '--module' flag is provided.
even though I have AMD checked in the project settings. 
Visual Studio 2015. 14.0.23107.0 D14REL
TypeScript Tools for Microsoft Visual Studio 2015 1.5.4.0

**Comment by danquirk** [comment]
What project type? Can you share the project file settings associated with Typescript and the actual compiler invocation from the output window?

**Comment by flyfishMT** [comment]
It is a .NET FW 4.6 Web Application, Empty Template. Not much in the output window: 

1>------ Build started: Project: xxx, Configuration: Debug Any CPU ------
1>C:\xxx\app\path\to\xxx\xxx.ts(16,1): error TS1148: Build: Cannot compile modules unless the '--module' flag is provided.
========== Build: 0 succeeded, 1 failed, 3 up-to-date, 0 skipped ==========

The code it is complaining about looks like this -
     class xxx {
     }
     export var viewModel: xxx;

**Comment by flyfishMT** [comment]
From project file:
<PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|Any CPU'">
    <TypeScriptTarget>ES5</TypeScriptTarget>
    <TypeScriptCompileOnSaveEnabled>True</TypeScriptCompileOnSaveEnabled>
    <TypeScriptNoImplicitAny>False</TypeScriptNoImplicitAny>
    <TypeScriptModuleKind>AMD</TypeScriptModuleKind>
    <TypeScriptRemoveComments>False</TypeScriptRemoveComments>
    <TypeScriptOutFile />
    <TypeScriptOutDir />
    <TypeScriptGeneratesDeclarations>False</TypeScriptGeneratesDeclarations>
    <TypeScriptNoEmitOnError>True</TypeScriptNoEmitOnError>
    <TypeScriptSourceMap>True</TypeScriptSourceMap>
    <TypeScriptMapRoot />
    <TypeScriptSourceRoot />
  </PropertyGroup

**Comment by mhegazy** [comment]
looks like a dupe of #4077

the project property pages saved the property changes under a wrong platform name "Any CPU" with a space in the middle. for now, just edit the file manually, change `"'$(Configuration)|$(Platform)' == 'Debug|Any CPU'"` to `"'$(Configuration)' == 'Debug'">` and it should work again.

**Comment by flyfishMT** [comment]
I made this change.  When you open the project settings in VS, "None" is checked as the module system. Change to AMD, save, same error. It adds a duplicate settings section just like I pasted, so it is ignoring the one I manually changed.

**Comment by mhegazy** [comment]
can you share more details? are your changes before or after the import to "Microsoft.TypeScript.targets"? it has to be before it to be picked up by the targets.

**Comment by flyfishMT** [comment]
I did not move anything around, just removed the |Any Cpu text as recommended.  It is a stock proj file otherwise. The section in question is directly before the Microsoft.TypesScript.Targets line. To illustrate what I was describing, here's what that part looks like after - deleting |Any CPU, opening the project settings in VS, setting it back to AMD, and saving:
```
  <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug'">
    <TypeScriptTarget>ES5</TypeScriptTarget>
    <TypeScriptCompileOnSaveEnabled>True</TypeScriptCompileOnSaveEnabled>
    <TypeScriptNoImplicitAny>False</TypeScriptNoImplicitAny>
    <TypeScriptModuleKind>AMD</TypeScriptModuleKind>
    <TypeScriptRemoveComments>False</TypeScriptRemoveComments>
    <TypeScriptOutFile />
    <TypeScriptOutDir />
    <TypeScriptGeneratesDeclarations>False</TypeScriptGeneratesDeclarations>
    <TypeScriptNoEmitOnError>True</TypeScriptNoEmitOnError>
    <TypeScriptSourceMap>True</TypeScriptSourceMap>
    <TypeScriptMapRoot />
    <TypeScriptSourceRoot />
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|Any CPU'">
    <TypeScriptTarget>ES5</TypeScriptTarget>
    <TypeScriptCompileOnSaveEnabled>True</TypeScriptCompileOnSaveEnabled>
    <TypeScriptNoImplicitAny>False</TypeScriptNoImplicitAny>
    <TypeScriptModuleKind>AMD</TypeScriptModuleKind>
    <TypeScriptRemoveComments>False</TypeScriptRemoveComments>
    <TypeScriptOutFile />
    <TypeScriptOutDir />
    <TypeScriptGeneratesDeclarations>False</TypeScriptGeneratesDeclarations>
    <TypeScriptNoEmitOnError>True</TypeScriptNoEmitOnError>
    <TypeScriptSourceMap>True</TypeScriptSourceMap>
    <TypeScriptMapRoot />
    <TypeScriptSourceRoot />
  </PropertyGroup>
  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\TypeScript\Microsoft.TypeScript.targets" Condition="Exists('$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\TypeScript\Microsoft.TypeScript.targets')" />
  <Import Project="$(VSToolsPath)\WebApplications\Microsoft.WebApplication.targets" Condition="'$(VSToolsPath)' != ''" />
  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v10.0\WebApplications\Microsoft.WebApplication.targets" Condition="false" />
```

Maybe changing it to AnyCPU instead of deleting might be a workaround?

**Comment by mhegazy** [comment]
it should be:
```xml
<PropertyGroup Condition="'$(Configuration)' == 'Debug'">
```

instead of:

```xml 
<PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug'">
```

**Comment by mhegazy** [comment]
changing it to `AnyCPU` with no spaces, will work for build, but you will not get the correct properties in the project properties.

**Comment by flyfishMT** [comment]
Yes, I missed the platform part. This workaround works thank you.

**Comment by mhegazy** [comment]
sorry for the hassle. hopefully this will be fixed soon.
